tests/foreman/cli/test_lifecycleenvironment.py

```diff
@@ -68,64 +68,47 @@ def test_bugzilla_1077333(self):
         self.assertEqual(result['name'], test_data['name'])
 
     # CRUD
-    @data(
-        {'name': gen_string("alpha", 15)},
-        {'name': gen_string("alphanumeric", 15)},
-        {'name': gen_string("numeric", 15)},
-        {'name': gen_string("latin1", 15)},
-        {'name': gen_string("utf8", 15)},
-        {'name': gen_string("html", 15)},
-    )
-    def test_positive_create_1(self, test_data):
-        """@Test: Create lifecycle environment with valid name, prior to Library
+    def test_positive_create_1(self):
+        """@Test: Create lifecycle environment with valid name, prior to
+        Library
 
         @Feature: Lifecycle Environment
 
         @Assert: Lifecycle environment is created with Library as prior
 
         """
-        lifecycle_environment = make_lifecycle_environment({
-            'name': test_data['name'],
-            'organization-id': self.org['id'],
-        })
-        self.assertEqual(
-            lifecycle_environment['prior-lifecycle-environment'], ENVIRONMENT)
-
-    @data(
-        {'name': gen_string("alpha", 15)},
-        {'name': gen_string("alphanumeric", 15)},
-        {'name': gen_string("numeric", 15)},
-        {'name': gen_string("latin1", 15)},
-        {'name': gen_string("utf8", 15)},
-        {'name': gen_string("html", 15)},
-    )
-    def test_positive_create_2(self, test_data):
-        """@Test: Create lifecycle environment with valid name and description,
-        prior to Library
+        for name in valid_data_list():
+            with self.subTest(name):
+                lc_env = make_lifecycle_environment({
+                    'name': name,
+                    'organization-id': self.org['id'],
+                })
+                self.assertEqual(
+                    lc_env['prior-lifecycle-environment'], ENVIRONMENT)
+
+    def test_positive_create_2(self):
+        """@Test: Create lifecycle environment with valid description prior to
+        Library
 
         @Feature: Lifecycle Environment
 
         @Assert: Lifecycle environment is created with Library as prior
 
         """
-        description = test_data['name']
-        lifecycle_environment = make_lifecycle_environment({
-            'description': description,
-            'name': test_data['name'],
-            'organization-id': self.org['id'],
-        })
-        self.assertEqual(lifecycle_environment['name'], test_data['name'])
-        self.assertEqual(
-            lifecycle_environment['description'], description)
-        self.assertEqual(
-            lifecycle_environment['prior-lifecycle-environment'], ENVIRONMENT)
-
-    @data(
-        {'label': gen_string("alpha", 15)},
-        {'label': gen_string("alphanumeric", 15)},
-        {'label': gen_string("numeric", 15)},
-    )
-    def test_create_lifecycle_environment_by_label(self, test_data):
+        for desc in valid_data_list():
+            name = gen_alphanumeric()
```

Comment:
Not required, the env name is already tested in previous tests.  Here the intent is to test different values of desc only.

Also I am just curious if that for statement will work?
```sh
>>> for (i, j) in ([1, 2, 3], 0):
...   print i, j
... 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: too many values to unpack
```